### DISCUSSION TITLE
How to tell if airsim is using cpu or gpu

### DISCUSSION THREAD

**Discussion by CoolCoder54323** [microsoft/AirSim#3287] [Support Q&A]
When I run my model device is set to cuda, but then it says using cpu device does that mean it is not using the gpu?
![image](https://user-images.githubusercontent.com/29154547/103972668-69824f00-513b-11eb-8cfa-5bb00e6772ff.png)

**Comment by rajat2004** [comment]
This is dependent on your code, you're using a script which is not one of the examples in the repo, so can't say anything about that. For TensorFlow or PyTorch, you can switch the device for the model, lots of documentation and examples are there for that.
In a general sense, to check if GPU is begin used, you can use `nvidia-smi` and check the utilization and processes. `htop` is useful for CPU and memory metrics.
